pkg/providers/instance/metrics.go

```diff
@@ -0,0 +1,93 @@
+/*
+Portions Copyright (c) Microsoft Corporation.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package instance
+
+import (
+	coremetrics "github.com/Azure/karpenter-provider-azure/pkg/metrics"
+	"github.com/prometheus/client_golang/prometheus"
+	crmetrics "sigs.k8s.io/controller-runtime/pkg/metrics"
+)
+
+const (
+	instanceSubsystem = "instance"
+
+	imageID   = "image"
+	errorCode = "error"
+)
+
+var (
+	// VMCreateStartMetric tracks when VM creation starts.
+	//
+	// STABILITY: ALPHA - This metric may change or be removed without notice.
+	VMCreateStartMetric = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Namespace: coremetrics.Namespace,
+			Subsystem: instanceSubsystem,
+			Name:      "vm_create_start_total",
+			Help:      "Total number of VM creation operations started.",
+		},
+		[]string{imageID},
```

Comment:
Does ImageID contain OS as well? Wondering if we want that split out for easy azlinux vs ubuntu comparisons?

I know when Charlie added this we were only using it internally but I think we need to think more carefully about what we want here and what we can have.

Things like:
- VM size
- Zone
- Capacity type (spot vs dedicated)
- Network details (is this always the same for a whole cluster? If so, maybe don't need it)

Seem to make a lot of sense here and are probably OK from a cardinality perspective? Can we go over all the fields that we _could_ have and pick out the ones we decide we want/need?